src/linear_algebra/gram_schmidt_ortho.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao, Kevin Buzzard
+-/
+
+import tactic.basic
+import algebra.big_operators.basic
+import analysis.inner_product_space.basic
+import analysis.normed_space.is_R_or_C
+
+/-!
+# Gram-Schmidt Orthogonalization and Orthonormalization
+
+In this file we introduce Gram-Schmidt Orthogonalization and Orthonormalization
+
+## Main results
+
+- `proj` : projection between two vectors in the inner product space
+- `GS`   : Gram-Schmidt Process
+- `GS_Orthogonal` : the proof that "GS" produces an orthogonal system of vectors
+- `GS_unit` : Normalized "Gram-Schmidt" (i.e each vector in this system has unit length)
+- `GS_Orthornormal` : the proof that "GS_unit" produces an orthornormal system of vectors
+-/
+
+open_locale big_operators
+
+variables (𝕜 : Type*) (E : Type*) [is_R_or_C 𝕜] [inner_product_space 𝕜 E]
+
+/-- projection in the inner product space -/
+def proj (u v : E) : E := ((inner u v) / (inner u u) : 𝕜) • u
+
+/-- Definition of Gram-Schmidt Process -/
+def GS (f : ℕ → E) : ℕ → E
+| n := f n - ∑ i in finset.range(n),
+  if h1 : i < n then proj 𝕜 E (GS i) (f n) else f 37
```

Comment:
`GS` is not a very informative name in the grand scheme of things. A name like `gram_schmidt_sequence` or `gram_schmidt_process` would be much more appropriate: it's much easier to search for in the docs or to guess while working in VSCode. The same applies to most of the names in this file.

Reply:
Thank you! I have fixed it